src/runtime.rs

```diff
@@ -307,6 +328,20 @@ impl Class {
 
     }
 
+    /// Checks whether this class conforms to the specified protocol.
+    pub fn conforms_to_protocol(&self, proto: &Protocol) -> bool {
+        unsafe { class_conformsToProtocol(self, proto) == YES }
+    }
+
+    /// Get a list of the protocols to which this class conforms.
+    pub fn protocols(&self) -> MallocBuffer<&Protocol> {
```

Comment:
👍 That sounds like a solid idea. When I was writing it, I did feel uneasy about the difference between `Class::protocols` and `Protocol::protocols`. This is better, and in fact, I'll add `Protocol::adopted_protocols` as part of this commit